docs/Code-Organization.md

```diff
@@ -0,0 +1,102 @@
+
+Three maven modules make up the central stack of the new web API implementation.
+
+* [web](../web) : define Web API request handlers, map endpoints, select services
+* [service](../service) : call persistence module or utils, apply business logic
+* [persistence](../persistence) : retrieve data from database. Two submodules:
+  * [persistence-api](../persistence/persistence-api) : declaration of repository classes / functions, caching markup
+  * [persistence-mybatis](../persistence/persistence-mybatis) : implementation of persistence-api using [MyBatis](https://mybatis.org/mybatis-3/) mappers
+
+#### Other Modules
+
+* [model](../model) : data model POJO classes (used throughout the stack)
+* [security](../security) : user authentication methods and authorization, request filters 
+* [business](../business) : a legacy refactor of webservice.do, has active dataSource definition
+* [portal](../portal) : web application packaging and launch
+* [scripts](../scripts) : data import tool packaging (from core module scripts package)
+* [db_scripts](../db_scripts) : installation and migration scripts for the database
+
+#### External Code Modules
+
+* [cbioportal-frontend](https://github.com/cBioPortal/cbioportal-frontend) : a React application using MobX and TypeScript
+* [session-service](https://github.com/cBioPortal/session-service) : an external session key/query specifier storage system
+
+**cbioportal-frontend** is packaged in the web application as a default frontend
+implementation, but the source of the frontend code can also be directed to an external
+source host and be deployed independently of the backend web application. See
+[details](Deployment-Procedure.md)
+
+**session-service** is imported into the web module in order to set up a proxy service
+which receives and forwards requests for saved cBioPortal sessions to a separate system
+providing this storage (using a document based database) The code is needed for handling
+modeled types such as VirtualStudy and Session
+
+### cBioPortal Module Dependencies
+
+![Module Dependencies](images/maven-module-dependencies.png)
+
+_A module is directly dependent on all modules which touch it from below (touching only at a corner
+does not count). Dependencies are transitive : web is dependent on service, service
+is dependent on persistence-mybatis, therefore web is also dependent on
+persistence-mybatis. During packaging, all dependencies are made available to the
+dependent package for use. Avoid creating new dependencies between modules.
+**cyclical dependencies will cause errors during build**_
+
+Module | Dependencies
+------ | ------------
+portal | core, _cbioportal-frontend_
+scripts | core
+core | web 
+web | business, service, _session-service_
+security-spring | service
+service | persistence-mybatis
+persistence-mybatis | persistence-api, db-scripts
+persistence-api | model
+business | model
+db-scripts |
+model |
+
+### Backend Development Guidelines
```

Comment:
I would pull this out and place it in a separate file.

Reply:
The README.md has a link to CONTRIBUTING.md, so I have separated these guidelines into their own file and put a link in CONTRIBUTING.md in the `Getting Started` section.

Reply:
I also put a direct link in the README.txt